### PR TITLE
PySpark dataframe related tests

### LINKED ISSUE #5881: [CT-1206] [Enhancement] test for saving pandas dataframe in python models
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Add a test in basic adapter tests for python model to help testing save pandas dataframe as this becomes a standard thing that we would want to support
Related link 
https://github.com/dbt-labs/dbt-bigquery/pull/301
https://github.com/dbt-labs/dbt-spark/pull/469

### Describe alternatives you've considered

_No response_

### Who will this benefit?

_No response_

### Are you interested in contributing this feature?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ChenyuLInx**
resolves #5881

### Description
Add tests related to pyspark dataframes
Run this in both dbt-spark and dbt-bigquery already, will open PR once this merged

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR

### CODE REVIEW

**Review by McKnight-42 — APPROVED** [review]
seems solid!

File: tests/adapter/dbt/tests/adapter/python_model/test_spark.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        'Latitude': [-34.58, -15.78, -33.45, 4.60, 10.48],
+        'Longitude': [-58.66, -47.91, -70.66, -74.08, -66.86]}
+        )
+
+    return df
+"""
+PYSPARK_MODEL = """
+# models/pyspark_df.py
+
+def model(dbt, session):
```
Comment (dbeatty10) [inline tests/adapter/dbt/tests/adapter/python_model/test_spark.py:24]: Maybe drop the first two lines so all three models have the same format?
```suggestion
def model(dbt, session):
```

File: tests/adapter/dbt/tests/adapter/python_model/test_spark.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        ],
+        ["City", "Country", "Latitude", "Longitude"]
+    )
+
+    return df
+"""
+
+PANDAS_ON_SPARK_MODEL = """
+
+import pyspark.pandas as ps
```
Comment (dbeatty10) [inline tests/adapter/dbt/tests/adapter/python_model/test_spark.py:45]: Maybe drop the first (empty) line so all three models have the same format?
```suggestion
import pyspark.pandas as ps
```

**Review by dbeatty10 — APPROVED** [review]
Looks great. Left comments for two nit-picky things.

**Review by dbeatty10 — APPROVED** [review]
Ready to merge from my perspective 🏆

**Pull request merged**